android/engine/src/main/java/org/smartregister/fhircore/engine/util/PractitionerDetailsUtils.kt

```diff
@@ -0,0 +1,159 @@
+      ResourceType.CareTeam.name ->
+        return getResourcesList(
+          practitionerDetails,
+          ResourceType.CareTeam.name,
+          CareTeam::class.java
+        )
+      ResourceType.Organization.name ->
+        return getResourcesList(
+          practitionerDetails,
+          ResourceType.Organization.name,
+          Organization::class.java
+        )
+      ResourceType.Location.name ->
+        return getResourcesList(
+          practitionerDetails,
+          ResourceType.Location.name,
+          Location::class.java
+        )
+      else -> arrayListOf()
+    }
+  }
+
+  suspend fun <T : Resource> getResourcesList(
+    practitionerDetails: Parameters,
+    resourceName: String,
+    clazz: Class<T>
+  ): ArrayList<T> {
+    val resourceList = arrayListOf<T>()
+    practitionerDetails.parameter.forEach {
+      if (it.name.equals(resourceName)) {
+        val result = it.resource as ListResource
+        if (result.hasEntry()) {
+          result.entry.forEach { entry ->
+            resourceList.add(
+              fhirEngine.load(
+                clazz = clazz,
+                id = entry.item.reference.replace("$resourceName/", "")
+              )
+            )
+          }
+        }
+      }
+    }
+    return resourceList
+  }
+
+  fun saveParameter(
+    practitionerId: String,
+    careTeamList: List<CareTeam>,
+    organizationList: List<Organization>,
+    locationList: List<Location>,
+  ) {
+    val parameters = Parameters()
+    parameters.addParameter().apply {
+      name = ResourceType.Practitioner.name
+      value = StringType(practitionerId)
+    }
+    addParameters(careTeamList, parameters, ResourceType.CareTeam.name)
+    addParameters(organizationList, parameters, ResourceType.Organization.name)
+    addParameters(locationList, parameters, ResourceType.Location.name)
+
+    sharedPreferences.write(
+      PRACTITIONER_PARAMETERS_SHARED_PREFERENCE_KEY,
+      parameters.encodeResourceToString()
+    )
+  }
+
+  fun <T : Resource> addParameters(
+    resources: List<T>,
+    parameters: Parameters,
+    resourceName: String
+  ) {
+    if (resources.isNotEmpty())
+      parameters.addParameter().apply {
+        name = resourceName
+        resource =
+          ListResource().apply {
+            resources.forEach { addEntry().apply { item = it.asReference() } }
+          }
+      }
+  }
+
+  fun storeUserPreferences(userInfo: UserInfo) {
+    sharedPreferences.write(USER_INFO_SHARED_PREFERENCE_KEY, userInfo.encodeJson())
+  }
+
+  fun updateUserDetailsFromPractitionerDetails(
+    practitionerDetails: PractitionerDetails,
+    userResponse: UserInfo
+  ) {
+    storeUserPreferences(
+      userInfo =
+        UserInfoItemMapper().mapToDomainModel(practitionerDetails, domainModelSource = userResponse)
+    )
+  }
+
+  fun storeKeyClockInfo(practitionerDetails: PractitionerDetails) {
+    val userData = practitionerDetails.userDetail as KeycloakUserDetails
+    val gson = Gson()
```

Comment:
Gson instance is provided via DI can be used here. Any reason for not using kotlin serialization?